modules/text-input/react/lib/TextInput.tsx

```diff
@@ -59,7 +59,28 @@ const Input = styled('input')<Pick<TextInputProps, 'error' | 'grow'>>(
   ({grow}) =>
     grow && {
       width: '100%',
-    }
+    },
+  ({theme, error}) => {
+    return {
+      '&:hover': {
+        borderColor: theme.palette.neutral.main,
+      },
+      '&:focus:not([disabled])': {
+        borderColor: theme.palette.common.focusOutline,
+        boxShadow: `inset 0 0 0 1px ${theme.palette.common.focusOutline}`,
+        outline: 'none',
+      },
+      '&:disabled': {
+        backgroundColor: theme.palette.neutral.lightest,
```

Comment:
Another thought about this is how do we communicate to the consumer when they're using a custom theme, that they need to use this specific property `theme.palette.neutral.lightest` to be able to override this